src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java

```diff
@@ -328,7 +323,7 @@ public <Z> Type map(TypeMapping<Z> mapping) {
      *  and with given constant value
      */
     public Type constType(Object constValue) {
```

Comment:
so now any type can have a constant value?

Reply:
Yeah - not sure which way to go - in reality we have a also a couple of types which implement this method and just return `this`. Only the toplevel `Type` declaration throws. I'm happy to revert to the old ways if that is preferred - it was mostly a way not to duplicate the method body twice (but that's not a big deal).

Reply:
I would keep the original semantics and we can relax it later if needed, but I don't have a strong preference tbh